mongo/index_view.go

```diff
@@ -397,21 +403,42 @@ func (iv IndexView) drop(ctx context.Context, name string, opts ...*options.Drop
 
 	// TODO(GODRIVER-3038): This operation should pass CSE to the DropIndexes
 	// Crypt setter to be applied to the operation.
-	op := operation.NewDropIndexes(name).
-		Session(sess).WriteConcern(wc).CommandMonitor(iv.coll.client.monitor).
-		ServerSelector(selector).ClusterClock(iv.coll.client.clock).
-		Database(iv.coll.db.name).Collection(iv.coll.name).
-		Deployment(iv.coll.client.deployment).ServerAPI(iv.coll.client.serverAPI).
-		Timeout(iv.coll.client.timeout).MaxTime(dio.MaxTime)
 
-	err = op.Execute(ctx)
-	if err != nil {
-		return nil, replaceErrors(err)
+	var results operation.DropIndexesResult
+
+	switch indexVal := index.(type) {
```

Comment:
The generic solution is a good start for the investigation into GODRIVER-3262. However, the current syntax is a bit awkward given that we need to maintain this switch condition. Suggest deferring the generic conversion to 3.0 as part of  GODRIVER-3262 and for now just use an `any` type. This would allow us to remove the need for this switch condition.

```go
type DropIndexes struct {
	index        any
}

func NewDropIndexes(index any) *DropIndexes {
	return &DropIndexes{
		index: index,
	}
}

func (di *DropIndexes) command(dst []byte, _ description.SelectedServer) ([]byte, error) {
	dst = bsoncore.AppendStringElement(dst, "dropIndexes", di.collection)

	switch any(di.index).(type) {
	case string:
		dst = bsoncore.AppendStringElement(dst, "index", di.index.(string))
	case bsoncore.Document:
		dst = bsoncore.AppendDocumentElement(dst, "index", di.index.(bsoncore.Document))
	}

	return dst, nil
}
```